python/doc/examples/numerical_methods/general_methods/plot_estimate_integral_iterated_quadrature.py

```diff
@@ -1,48 +1,272 @@
+# function :math:`f`. We can see that the algorithm focuses on the nodes where the function
+# has fast variations.
+cloud_nodes = ot.Cloud(nodes)
+cloud_nodes.setPointStyle('dot')
+cloud_nodes.setColor('blue')
+
+g.add(cloud_nodes)
+g.setLegendPosition('')
+view = otv.View(g)
+
+
+# %%
+# Case 3: The integration domain is complex.
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# We consider the function:
+#
+# .. math::
+#
+#    f : \begin{array}{lcl}
+#          \Rset^2 & \rightarrow & \Rset \\
+#          (x,y) & \rightarrow & cos(2x) * sin(1.5y)
+#        \end{array}
+#
+# the domain :math:`\set{D} = \{(x,y)\, |\, -2\pi \leq x \leq 2\pi, -2-cos(x) \leq y  \leq 2+cos(x) \}`
+# and the integral:
+#
+# .. math::
+#    I = \int_{\set{D}} f(x,y)\,dxdy.
+
+# %%
+# We first define the integrand :math:`f` and the domain :math:`\set{D}`.
+
+
+def kernel3(x):
+    return [m.cos(2.0 * x[0]) * m.cos(1.5 * x[1])]
+
+
+integrand = ot.PythonFunction(2, 1, kernel3)
+
+# %%
+# The domain of integration is defined by the two following functions:
+#
+# .. math::
+#
+#    x \rightarrow u(x) = 2+cos(x)
+#
+# .. math::
+#
+#    x \rightarrow  \ell(x) = -2-cos(x)
+#
+upper_func = ot.SymbolicFunction(["x"], [" 2 + cos(x)"])
+lower_func = ot.SymbolicFunction(["x"], ["-2 - cos(x)"])
+
+# %%
+# We draw the iso-lines of the integrand function in the domain of integration.
+# To do that, we define a new function which is the restriction
+# of the integrand function to the integration domain.
+
+
+def kernel3_insideDomain(x):
+    low_x = lower_func([x[0]])[0]
+    up_x = upper_func([x[0]])[0]
+    if x[1] > low_x and x[1] < up_x :
+        return kernel3(x)
+    else:
+        return [0.0]
+
+
+integrand_domain = ot.PythonFunction(2, 1, kernel3_insideDomain)
+a = 2 * m.pi
+b = 4.0
+# here, we ask for more levels than the default number store in :class:`~openturns.ResourceMap`.
+ot.ResourceMap.SetAsUnsignedInteger("Contour-DefaultLevelsNumber", 30)
+g = integrand_domain.draw([-a, -b], [a, b])
+
+# %%
+# We add the bounds of the domain :math:`\set{D}`.
+low_bound = lower_func.draw([-a, -b], [a, b]).getDrawable(0)
+up_bound = upper_func.draw([-a, -b], [a, b]).getDrawable(0)
+low_bound.setColor('red')
+low_bound.setLineWidth(2)
+up_bound.setColor('red')
+up_bound.setLineWidth(2)
+
+g.add(low_bound)
+g.add(up_bound)
+g.setTitle(r'Function $f(x,y) =  cos(2x) * sin(1.5y)$')
+g.setXTitle('x')
+g.setYTitle('y')
+view = otv.View(g)
+
+# %%
+# To get the nodes used to approximate the integral, we need to use a :class:`~openturns.MemoizeFunction`
+# that stores all the calls to the function.
+integrand_memoized = ot.MemoizeFunction(integrand)
+maximumSubIntervals = 4
+maximumError = 1e-4
+GKRule = ot.GaussKronrodRule(ot.GaussKronrodRule.G7K15)
+integration_algo = ot.IteratedQuadrature(ot.GaussKronrod(maximumSubIntervals, maximumError, GKRule))
+I_value = integration_algo.integrate(integrand_memoized, -a, a, [lower_func], [upper_func])
+print('I = ', I_value[0])
```

Comment:
The exact value is -0.548768615494004063851543284777 according to Maple

Reply:
ok